utils/error_injection.hh

```diff
@@ -43,6 +43,11 @@ extern logging::logger errinj_logger;
 
 using error_injection_parameters = std::unordered_map<sstring, sstring>;
 
+struct wait_for_message {
+    std::chrono::milliseconds timeout;
+    wait_for_message(std::chrono::milliseconds tmo) noexcept : timeout(tmo) {}
```

Comment:
I think the intent was not to hang the whole thing in case test does something wrong (i.e. -- doesn't wake up scylla with the message). I agree it now looks weird, especially the selection of timeout values themselves

Reply:
I can understand that we want a test **failure**, especially an expected failure (xfail), to report a failure in a reasonable time.  But why do the 20 different tests that use this feature need to invent what this "reasonable time" is? I saw various numbers between 10 seconds and 5 minutes. I suggested that:
1. We can have a default and explain that this is supposed to be "infinite" but was actually set to 5 minutes or 1 minute or whatever to allow faster failure.
2. If there is **any** case where we need this timeout to be lower, we need to explain what is the reason. So far I'm not sure that there is **any** case where the default won't do. Is there?

Finally, there's still something I don't understand. It seems to me this timeout is just about the case of a buggy test, that "forgets" to release the injection. How/why does this happen without the test having failed first? In some frameworks, when a test fails, Scylla is aborted so we don't care about the injection. But in other frameworks (e.g., cql-pytest) Scylla is not aborted so we need to release the injection in a guaranteed way, using try/finally or context manager - so again the timeout isn't necessary or even helpful (we don't want the injection to be stuck for 5 minutes if we forgot to release it).